pkg/types/service_binding.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+// Package types contains the Service Manager web entities
+package types
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/Peripli/service-manager/pkg/util"
+)
+
+//go:generate smgen api ServiceBinding
+// ServiceBinding struct
+type ServiceBinding struct {
+	Base
+	Secured           `json:"-"`
+	Name              string          `json:"name"`
+	ServiceInstanceID string          `json:"service_instance_id"`
+	SyslogDrainURL    string          `json:"syslog_drain_url,omitempty"`
+	RouteServiceURL   string          `json:"route_service_url,omitempty"`
+	VolumeMounts      json.RawMessage `json:"volume_mounts,omitempty"`
+	Endpoints         json.RawMessage `json:"endpoints,omitempty"`
+	Context           json.RawMessage `json:"-"`
+	BindResource      json.RawMessage `json:"-"`
+	Credentials       string          `json:"credentials"`
```

Comment:
May be it will be better for this to be of type RawMessage or []byte because we don't know what the broker will put there

Reply:
As discussed this is a limitation of the storage layer, because it cannot Scan into RawMessage. Also because it is encrypted in the database it cannot be any other db field, but byte array